In [159]:
import pandas as pd
import yfinance as yf

montantInitial = 10000
montantInvest = 10000


data = yf.download("TSLA")

data.head()

[*********************100%%**********************]  1 of 1 completed


Open      High       Low     Close  Adj Close     Volume
Date                                                                    
2010-06-29  1.266667  1.666667  1.169333  1.592667   1.592667  281494500
2010-06-30  1.719333  2.028000  1.553333  1.588667   1.588667  257806500
2010-07-01  1.666667  1.728000  1.351333  1.464000   1.464000  123282000
2010-07-02  1.533333  1.540000  1.247333  1.280000   1.280000   77097000
2010-07-06  1.333333  1.333333  1.055333  1.074000   1.074000  103003500

In [75]:
data.columns

Index(['Open', 'High', 'Low', 'Close', 'Adj Close', 'Volume'], dtype='object')

In [160]:
# Prétraitement nettoyage des données
# copie la dataset en un pandas dataframe
df = pd.DataFrame(data)

# on a pas des valeurs null
len(df[df["Open"] == None])


0

In [114]:
len(df[df["High"] == None])

0

In [43]:
len(df[df["Low"] == None])

0

In [44]:
len(df[df["Adj Close"] == None])

0

In [45]:
len(df[df["Volume"] == None])

0

# Exploration des données

## calcule de l'écart type et l'écart interquartille

In [115]:
# calcule d'ecart type
print("Le prix moyen des prix d'ouverture ", df["Open"].mean())

df["Open"].std()

Le prix moyen des prix d'ouverture  69.998719360441


101.23716516801491

**Remarque:** Pour les prix d'ouver[link text](https://)ture a une variation de 101.20\$ par rapport à la moyenne  qui est à 69.94\$

In [61]:
# calcule de l'ecart interquartille
quartiles = df["Open"].quantile([0.25, 0.75])
print(f"Q3 = {quartiles[0.75]}, Q1 = {quartiles[0.25]}")
iqr = quartiles[0.75] - quartiles[0.25]
print(iqr)

Q3 = 101.20050048828125, Q1 = 10.5
90.70050048828125


L'ecart interquartille : écart entre le premier quartille et le troisieme quartille est 90.7$

C-à-d on a une variabilité elevé de 90.7$ dans la dataset, une différence elevé par rapport à la moyenne 69.94\$

In [62]:
df["Close"].std()

101.14040434422425

In [63]:
df["Close"].mean()

69.9338440271957

**Remarque:** Pour les prix de fermeture a une variation de 101.147\$ par rapport à la moyenne  qui est à 69.93\$

In [64]:
# calcule de l'ecart interquartille
quartiles = df["Close"].quantile([0.25, 0.75])
print(f"Q3 = {quartiles[0.75]}, Q1 = {quartiles[0.25]}")
iqr = quartiles[0.75] - quartiles[0.25]
print(iqr)

Q3 = 102.26000022888184, Q1 = 10.675333738327026
91.58466649055481


L'ecart interquartille : écart entre le premier quartille et le troisieme quartille est 91.58$

C-à-d on a une variabilité elevé de 90.7\$ dans la dataset par rapport à la moyenne qui est à 69.93$ tel que dans le premier quartille les prix de

fermeture sont à 10.67\$ et dans le Q3 on a une 10.67\$

## identification des métriques clés

### Rendement *moyen*

In [161]:
# dans ce cas on va calculer le rendement annuel

# calcule de rendement

montantRecurant = 500

nbAction = 0

counter = 0

startYear = 2022
endYear = 2023

for ind in df.index :

  if ind.day == 1 and montantInvest >= montantRecurant and ind.year >= startYear and ind.year <= endYear:
    nbAction += montantRecurant / df["Adj Close"][ind]
    montantInvest -= montantRecurant

nbAction

lastAdjClose = df["Adj Close"][-1]
revenue = nbAction * lastAdjClose + montantInvest
revenue

rendementMoyen = (revenue / montantInitial)

print(f"Le rendement moyen entre {startYear} et {endYear} = {rendementMoyen}% {revenue}")

Le rendement moyen entre 2022 et 2023 = 1.058650008729988% 10586.500087299879
